### ISSUE TITLE
Plugin needs to change TOC.XML file

### ISSUE THREAD

**Issue by varnk** [EWSoftware/SHFB#74]
In my custom plug-in, I would like to change the default Table of Contents that is generated when I run SandCastle against a C# XML file created by Visual Studio from source code.  The default TOC creates a hierarchy of namespaces, classes, and members.  I would like to change the name and order of the TOC tree in the help that is generated.  I thought I could do this by modifying the TOC.XML file in my plug-in, but I am not sure how the TOC.XML file elements related to the TocEntry class members.  Is there any documentation on the format of the TOC.XML file available?  Or, is there some other way I should be doing this?

**Comment by EWSoftware** [comment]
TocEntry is just used to track information about additional content items (HTML files) which end up in _ConceptualTOC_.xml.  The toc.xml file is generated by the XSL transformation specified by the presentation style, typically CreateVSToc.xsl.  The two TOC files are then merged into one file before transforming the content into the help format-specific TOC.

The TOC files don't contain title information, just ID and filename references.  The format-specific tasks extract titles if needed from the topic files.  Some, like MS Help Viewer don't do that since the viewer handles the TOC.

If you want to change how the API TOC is generated, you may need to look into using an alternate XSL transformation and cloning the presentation style.

**Comment by varnk** [comment]
Turns out I was able to achieve my desired result by doing the following (note that this may not be the "right" way to do it, but it does work):

1.  Listen iin on BuildReferenceTopics BuildStep.
2.  Read in the TOC.XML file in the Execution method.
3.  Create root level TOC nodes with my own unique id name
4.  Shift the <topic> elements under the tree nodes that I created.

The resulting TOC.XML, looks like this snippet:

<?xml version="1.0" encoding="utf-8"?>
<topics>
  <topic id="TestClasses">
    <topic id="T:Test.DiagAnalogInput" project="ModuleTest" file="8266a1fd-0640-ea7a-359e-1989bda97ae1">
   </topic>
</topic>
</topics>

This seems to work, but then again maybe I just got lucky.
